tests/Api/ApiTestCase.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Akeneo\Pim\ApiClient\tests\Api;
+
+use Akeneo\Pim\ApiClient\AkeneoPimClientBuilder;
+use Akeneo\Pim\ApiClient\AkeneoPimClientInterface;
+use Akeneo\Pim\ApiClient\Api\AuthenticationApi;
+use donatj\MockWebServer\MockWebServer;
+use donatj\MockWebServer\Response;
+use donatj\MockWebServer\ResponseStack;
+use Http\Message\StreamFactory;
+use Http\Discovery\StreamFactoryDiscovery;
+
+/**
+ * @author    Laurent Petard <laurent.petard@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+abstract class ApiTestCase extends \PHPUnit_Framework_TestCase
+{
+    /** @var MockWebServer */
+    protected $server;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp()
+    {
+        $this->server = new MockWebServer(8081, '127.0.0.1');
+        $this->server->start();
+
+        $this->server->setResponseOfPath(
+            '/'. AuthenticationApi::TOKEN_URI,
+            new ResponseStack(
+                new Response($this->getAuthenticatedJson())
+            )
+        );
+    }
+
+    /**
+     * @return AkeneoPimClientInterface
+     */
+    protected function createClient()
+    {
+        $clientBuilder = new AkeneoPimClientBuilder($this->server->getServerRoot());
+
+        return $clientBuilder->buildAuthenticatedByPassword(
+            'client_id',
+            'secret',
+            'username',
+            'password'
+        );
+    }
+
+    private function getAuthenticatedJson()
+    {
+        return <<<JSON
+            {
+                "refresh_token" : "this-is-a-refresh-token",
+                "access_token" : "this-is-an-access-token"
+            }
+JSON;
+    }
+
+    /**
+     * @return StreamFactory
+     */
+    protected function getStreamFactory()
```

Comment:
where is it used?

Reply:
In UpsertListProductTest sur le test test_upsert_list_from_stream